performance_manager/lib/rt_vehicle_positions.py

```diff
@@ -286,49 +276,57 @@ def merge_vehicle_position_events(
                 ),
             )
 
+    process_logger.log_complete()
+
 
 def process_vehicle_positions(db_manager: DatabaseManager) -> None:
     """
     process a bunch of vehicle position files
     create events for them
     merge those events with existing events
     """
+    process_logger = ProcessLogger("process_vp")
+    process_logger.log_start()
 
     # check metadata table for unprocessed parquet files
     paths_to_load = get_unprocessed_files(
         "RT_VEHICLE_POSITIONS", db_manager.get_session()
     )
 
+    process_logger.add_metadata(paths_to_load=len(paths_to_load))
+
     for folder, folder_data in paths_to_load.items():
         ids = folder_data["ids"]
         paths = folder_data["paths"]
 
+        subprocess_logger = ProcessLogger(
+            "process_tu_dir", folder=folder, file_count=len(paths)
+        )
+        subprocess_logger.log_start()
+
         try:
             new_events = get_vp_dataframe(paths)
-            logging.info(
-                "Size of dataframe from %s is %d", folder, new_events.shape[0]
-            )
-            new_events = transform_vp_dtyes(new_events)
-            logging.info(
-                "Size of dataframe with updated dtypes is %d",
-                new_events.shape[0],
-            )
-            new_events = transform_vp_timestamps(new_events)
-            logging.info(
-                "Size of dataframe with transformed timestamps is %d",
-                new_events.shape[0],
+            new_events_updated = transform_vp_dtyes(new_events)
+            new_events_timestamps = transform_vp_timestamps(new_events_updated)
```

Comment:
same as with trip updates (but likely not as much of an issue) keeping all of these dataframe variables in scope at the same time could cause memory use issues

Reply:
same